base_update_restrict_model/tests/test_base_update_restrict_model.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.exceptions import AccessError
+from odoo.tests import SavepointCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestBaseUpdateRestrictModel(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner_model = cls.env["ir.model"].search([("model", "=", "res.partner")])[
+            0
+        ]
+        cls.partner_model.update_restrict = True
+        cls.test_partner = cls.env["res.partner"].create({"name": "Test Partner"})
+        cls.restrict_test_user = cls.env["res.users"].create(
+            {
+                "name": "Resticted user",
+                "login": "resticted@example.com",
+                "restrict_update_permit": False,
+            }
+        )
```

Comment:
Looks like Travis is complaining that there is no email for the partner. Would it make it simpler to just directly create the user without creating a partner beforehand. I suppose that email should be automatically updated with the login that way.